pkg/yurthub/locallb/iptables.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package locallb
+
+import (
+	"strconv"
+
+	"github.com/coreos/go-iptables/iptables"
+	"k8s.io/klog/v2"
+)
+
+type IptablesManager struct {
+	ipt iptables.IPTables
+}
+
+func NewIptablesManager() *IptablesManager {
+	iptable, _ := iptables.New()
+	m := &IptablesManager{
+		ipt: *iptable,
+	}
+	return m
+}
+
+func (im *IptablesManager) updateIptablesRules(cloudip string, ips []string) error {
+	if err := im.cleanIptablesRules(cloudip, ips); err != nil {
+		return err
+	}
+	if err := im.addIptablesRules(cloudip, ips); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (im *IptablesManager) addIptablesRules(cloudip string, ips []string) error {
+	ramdomBalancingProbability := im.getRamdomBalancingProbability(len(ips))
+	for index, ip := range ips {
+		// iptables规则需要修改
+		err := im.ipt.Append("nat", "PREROUTING", "-d", cloudip, "-p", "tcp", "--dport", "80", "-m", "statistic", "--mode", "random", "--probability", strconv.FormatFloat(ramdomBalancingProbability[index], 'f', -1, 64), "-j", "DNAT", "--to-destination", ip+":80")
```

Comment:
then we need to proxy the local traffic, so we need to use `OUTPUT` chain instead of `PREROUTING ` chain.

Reply:
done.